src/Main/Tasks.cpp

```diff
@@ -261,3 +262,49 @@ void FOEDAG::handleViewReportRequested(const QString& reportId,
 
   openReportView(*report);
 }
+
+void FOEDAG::TclArgs_setSimulateOptions(const std::string& argsStr) {
+  FOEDAG::Compiler* compiler = GlobalSession->GetCompiler();
+  if (!compiler) return;
+
+  std::vector<std::string> argsList;
+  StringUtils::tokenize(argsStr, " ", argsList, true);
+
+  if (argsList.size() < 2) return;
+
+  auto simulationType{argsList.at(0)};
+  auto waveFile{argsList.at(1)};
+
+  if (simulationType.compare("-rtl_filepath") == 0)
+    compiler->GetSimulator()->WaveFile(Simulator::SimulationType::RTL,
+                                       waveFile);
+  if (simulationType.compare("-gate_filepath") == 0)
+    compiler->GetSimulator()->WaveFile(Simulator::SimulationType::Gate,
+                                       waveFile);
+  if (simulationType.compare("-pnr_filepath") == 0)
+    compiler->GetSimulator()->WaveFile(Simulator::SimulationType::PNR,
+                                       waveFile);
+  if (simulationType.compare("-bitstream_filepath") == 0)
+    compiler->GetSimulator()->WaveFile(Simulator::SimulationType::Bitstream,
+                                       waveFile);
+}
+
+std::string FOEDAG::TclArgs_getSimulateOptions() {
+  FOEDAG::Compiler* compiler = GlobalSession->GetCompiler();
+  if (!compiler) return std::string{};
+
+  std::vector<std::string> argsList;
+  argsList.push_back("-rtl_filepath");
```

Comment:
Can the user leave the filename empty or there willl be a default name generated in  UI ?

Reply:
Yes he can. In this case name will be generate by the simulator based on top level module name. I didn't handle this for now.

Reply:
This isn't clear. Is it in a future PR ?

> I didn't handle this for now.

Reply:
Let's have a discussion on this

Reply:
This is resolved. I've added default values